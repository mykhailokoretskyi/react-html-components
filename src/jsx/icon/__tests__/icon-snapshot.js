jest.mock('react/lib/ReactDefaultInjection');
import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import TinyIcon from '../tinyIcon';
import SmallIcon from '../smallIcon';
import LargeIcon from '../largeIcon';
import MediumIcon from '../mediumIcon';

describe('Icon', () => {
    it('tiny icon snapshot', () => {
        const component = renderer.create(<TinyIcon iconName={"test_icon"} />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('small icon snapshot', () => {
        const component = renderer.create(<SmallIcon classes={["some test classes"]} iconName={"test_icon"} />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('medium icon snapshot', () => {
        const component = renderer.create(<MediumIcon classes={["some test classes"]} iconName={"test_icon"} />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('large icon matches snapshot', () => {
        const component = renderer.create(<LargeIcon classes={["some test classes"]} iconName={"test_icon"} />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
