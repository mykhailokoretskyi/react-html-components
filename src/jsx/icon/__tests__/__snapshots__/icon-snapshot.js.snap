exports[`Icon large icon matches snapshot 1`] = `
<i
  className="some test classes material-icons large"
  dangerouslySetInnerHTML={
    Object {
      "__html": "test_icon"
    }
  } />
`;

exports[`Icon medium icon snapshot 1`] = `
<i
  className="some test classes material-icons medium"
  dangerouslySetInnerHTML={
    Object {
      "__html": "test_icon"
    }
  } />
`;

exports[`Icon small icon snapshot 1`] = `
<i
  className="some test classes material-icons small"
  dangerouslySetInnerHTML={
    Object {
      "__html": "test_icon"
    }
  } />
`;

exports[`Icon tiny icon snapshot 1`] = `
<i
  className="material-icons tiny"
  dangerouslySetInnerHTML={
    Object {
      "__html": "test_icon"
    }
  } />
`;
